pkg/status/versionchecks.go

```diff
@@ -0,0 +1,60 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/google/go-github/v52/github"
+	errs "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// ErrMsgDeploymentIsNotUpToDate means that deployment version is not aligned with source code version
+	ErrMsgDeploymentIsNotUpToDate = "deployment version is not up to date with latest github commit SHA"
+
+	// DeploymentThreshold is the threshold after which we can be almost sure the deployment was not updated on the cluster with the latest version/commit,
+	// in this case some issue is preventing the new deployment to happen.
+	DeploymentThreshold = 30 * time.Minute
+)
+
+// CheckDeployedVersionIsUpToDate verifies if there is a match between the latest commit in Github for a given repo and branch matches the provided commit SHA.
+// There is some preconfigured delay/threshold that we keep in account before returning an `error condition`.
+func CheckDeployedVersionIsUpToDate(githubClient *github.Client, repoName, repoBranch, deployedCommitSHA string) *toolchainv1alpha1.Condition {
```

Comment:
just a side comment - returning the pointer is a bit weird because it never happens that you would return `nil` - the condition is always set, so you could return it as a value. But I guess that you wanted to align with the `NewComponentReadyCondition` which return a pointer as well. Let's keep it as it is for now :+1: